tests/DCPS/MultiDPTest/DataReaderListener.cpp

```diff
@@ -8,158 +8,125 @@
 #include "tests/DCPS/FooType5/FooDefTypeSupportImpl.h"
 
 template <class DT, class DT_seq, class DR, class DR_ptr, class DR_var>
-int read (::DDS::DataReader_ptr reader)
+int read(::DDS::DataReader_ptr reader)
 {
-  try
-  {
-    DR_var foo_dr
-      = DR::_narrow(reader);
-    if (CORBA::is_nil (foo_dr.in ()))
-    {
-      ACE_ERROR ((LM_ERROR,
-        ACE_TEXT("(%P|%t) read: _narrow failed.\n")));
-      throw TestException() ;
+  try {
+    DR_var foo_dr = DR::_narrow(reader);
+    if (CORBA::is_nil(foo_dr.in())) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) read: _narrow failed.\n")));
+      throw TestException();
     }
 
     DT foo;
-    ::DDS::SampleInfo si ;
+    ::DDS::SampleInfo si;
 
-    DDS::ReturnCode_t const status = foo_dr->read_next_sample(foo, si) ;
-
-    if (status == ::DDS::RETCODE_OK)
-    {
+    DDS::ReturnCode_t const status = foo_dr->read_next_sample(foo, si);
+    if (status == ::DDS::RETCODE_OK) {
       ACE_DEBUG((LM_DEBUG,
         ACE_TEXT("(%P|%t) reader %X foo.x = %f foo.y = %f, foo.data_source = %d \n"),
         reader, foo.x, foo.y, foo.data_source));
       ACE_DEBUG((LM_DEBUG,
         ACE_TEXT("(%P|%t) SampleInfo.sample_rank = %d \n"),
         si.sample_rank));
+    } else if (status == ::DDS::RETCODE_NO_DATA) {
+      ACE_ERROR_RETURN((LM_ERROR,
+        ACE_TEXT("(%P|%t) ERROR: reader received ::DDS::RETCODE_NO_DATA!\n")), -1);
+    } else {
+      ACE_ERROR_RETURN((LM_ERROR,
+        ACE_TEXT("(%P|%t) ERROR: read  foo: Error: %d\n"), status), -1);
```

Comment:
Isn't there a helper to log status as string?

Reply:
This is updated.